consensus/istanbul/validator/weighted.go

```diff
@@ -193,9 +194,18 @@ func NewWeightedCouncil(addrs []common.Address, rewards []common.Address, voting
 		valSet.validators[i] = newWeightedValidator(addr, rewards[i], votingPowers[i], weights[i])
 	}
 
-	// sort validator
+	// sort validators
 	sort.Sort(valSet.validators)
 
+	// init demoted validators
+	valSet.demotedValidators = make([]istanbul.Validator, len(demotedAddrs))
+	for i, addr := range demotedAddrs {
+		valSet.demotedValidators[i] = newWeightedValidator(addr, common.Address{}, 1000, 0)
+	}
+
+	// sort demoted validators
+	sort.Sort(valSet.demotedValidators)
```

Comment:
These lines can't be tested since there is no place that calls `NewWeightedCouncil` with non-nil `demotedAddrs`

Reply:
`NewWeightedCouncil` is only used properly while unmarshalling a snapshot. This is called when read a snapshot from database.